cpp/src/io/parquet/predicate_pushdown.cpp

```diff
@@ -444,55 +309,180 @@ std::optional<std::vector<std::vector<size_type>>> aggregate_reader_metadata::fi
+  _col_literals_offsets.emplace_back(0);
+
+  std::transform(equality_literals.begin(),
+                 equality_literals.end(),
+                 std::back_inserter(_col_literals_offsets),
+                 [&](auto const& col_literal_map) {
+                   return _col_literals_offsets.back() +
+                          static_cast<cudf::size_type>(col_literal_map.size());
+                 });
+
+  // Add this visitor
+  expr.accept(*this);
+}
+
+std::reference_wrapper<ast::expression const> combined_expression_converter::visit(
+  ast::literal const& expr)
+{
+  _converted_expr = std::reference_wrapper<ast::expression const>(expr);
+  return expr;
+}
+
+std::reference_wrapper<ast::expression const> combined_expression_converter::visit(
+  ast::column_reference const& expr)
+{
+  CUDF_EXPECTS(expr.get_table_source() == ast::table_reference::LEFT,
+               "CombinedAST (Stats and Bloomfilter) supports only left table");
+  CUDF_EXPECTS(expr.get_column_index() < _num_input_columns,
+               "Column index cannot be more than number of columns in the table");
+  _converted_expr = std::reference_wrapper<ast::expression const>(expr);
+  return expr;
+}
+
+std::reference_wrapper<ast::expression const> combined_expression_converter::visit(
+  ast::operation const& expr)
+{
+  using cudf::ast::ast_operator;
+  auto const operands = expr.get_operands();
+  auto const op       = expr.get_operator();
+
+  if (auto* v = dynamic_cast<ast::column_reference const*>(&operands[0].get())) {
+    // First operand should be column reference, second should be literal.
+    CUDF_EXPECTS(cudf::ast::detail::ast_operator_arity(op) == 2,
+                 "Only binary operations are supported on column reference");
+    CUDF_EXPECTS(dynamic_cast<ast::literal const*>(&operands[1].get()) != nullptr,
+                 "Second operand of binary operation with column reference must be a literal");
+    v->accept(*this);
+    auto const col_index = v->get_column_index();
+    switch (op) {
+      /* transform to stats conditions. op(col, literal)
+      col1 == val --> vmin <= val && vmax >= val && bloom_filter[col1, val].contains(val)
+      col1 != val --> !(vmin == val && vmax == val)
+      col1 >  val --> vmax > val
+      col1 <  val --> vmin < val
+      col1 >= val --> vmax >= val
+      col1 <= val --> vmin <= val
+      */
+      case ast_operator::EQUAL: {
+        auto const& vmin = _col_ref.emplace_back(col_index * 2);
+        auto const& vmax = _col_ref.emplace_back(col_index * 2 + 1);
+        auto const& op1 =
+          _operators.emplace_back(ast_operator::LESS_EQUAL, vmin, operands[1].get());
+        auto const& op2 =
+          _operators.emplace_back(ast_operator::GREATER_EQUAL, vmax, operands[1].get());
+        _operators.emplace_back(ast::ast_operator::LOGICAL_AND, op1, op2);
+
+        // Use this input column's bloom filter membership column as well if available.
+        if (_has_bloom_filters) {
```

Comment:
Handling for BloomfilterAST columns appended at the end of the StatsAST columns for `EQUAL` operator only.